src/components/Common/SearchInput.tsx

```diff
@@ -100,22 +101,34 @@ const SearchInputFieldRenderer = ({
 
         // Only call onSearch if the phone number is valid
         if (isValid) {
-          onSearch(selectedOption.key, phoneValue);
+          if (selectedOption.onSearch) {
+            selectedOption.onSearch(phoneValue);
+          } else {
+            onSearch(selectedOption.key, phoneValue);
+          }
```

Comment:
shouldn't both onSearch's get triggered? 🤔 
```suggestion
          selectedOption.onSearch?.(phoneValue);
          onSearch(selectedOption.key, phoneValue);
```

Reply:
The onSearch (in the props) is like a fallback, when there is no onSearch in the selectedOptions it falls back to the root one.

If it is triggered every time, patient identifier search calls would be made which would result in 404. These would be unnecessary api calls.